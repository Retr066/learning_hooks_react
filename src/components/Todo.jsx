import React from "react";

export default function Todo({ todo, handleDelete, handleToogleCompleted }) {
  const { id, title, desc, completed } = todo;

  return (
    <>
      <li className="list-group-item d-flex justify-content-between align-items-start">
        <div className="ms-2 me-auto">
          <div
            className={`fw-bold ${completed && "text-decoration-line-through"}`}
          >
            {title}
          </div>
          <span className={`${completed && "text-decoration-line-through"}`}>
            {desc}
          </span>
        </div>

        <input
          className="form-check-input p-3 bg-primary rounded-pill  "
          style={{ cursor: "pointer" }}
          type="checkbox"
          onChange={() => handleToogleCompleted(id)}
          checked={completed}
          id="flexCheckChecked"
        />
        <button
          className={`btn ms-3 btn-outline-danger  ${
            !completed && "visually-hidden"
          }`}
          onClick={() => handleDelete(id)}
        >
          Delete
        </button>
      </li>
    </>
  );
}
