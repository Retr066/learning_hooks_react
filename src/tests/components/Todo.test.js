import { shallow } from "enzyme";
import Todo from "../../components/Todo";
import { demoTodos } from "../fixtures/demoTodos";
import "@testing-library/jest-dom";

describe("Pruebas para el componente de Todo", () => {
  const handleDelete = jest.fn();
  const handleToogleCompleted = jest.fn();
  let wrapper = shallow(
    <Todo
      todo={demoTodos[0]}
      handleDelete={handleDelete}
      handleToogleCompleted={handleToogleCompleted}
    />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(
      <Todo
        todo={demoTodos[0]}
        handleDelete={handleDelete}
        handleToogleCompleted={handleToogleCompleted}
      />
    );
  });
  test("debe renderizar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe llamar ala function handleDelete", () => {
    wrapper.find("button").simulate("click");
    expect(handleDelete).toHaveBeenCalled();
    expect(handleDelete).toHaveBeenCalledTimes(1);
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("debe llamar ala function handleToogleCompleted", () => {
    wrapper.find("input").simulate("change");
    expect(handleToogleCompleted).toHaveBeenCalled();
    expect(handleToogleCompleted).toHaveBeenCalledTimes(1);
    expect(handleToogleCompleted).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("debe mostrar la description correctamente", () => {
    const text = wrapper.find("span").text().trim();
    expect(text).toBe("Study react hooks (useReducer)");
  });
  test("debe tener la clase text-decoration-line-through", () => {
    expect(wrapper.find("span").hasClass("text-decoration-line-through")).toBe(
      true
    );
  });
});
