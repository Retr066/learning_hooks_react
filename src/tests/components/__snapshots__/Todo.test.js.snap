// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente de Todo debe renderizar correctamente 1`] = `
<Fragment>
  <li
    className="list-group-item d-flex justify-content-between align-items-start"
  >
    <div
      className="ms-2 me-auto"
    >
      <div
        className="fw-bold text-decoration-line-through"
      >
        Learnig react
      </div>
      <span
        className="text-decoration-line-through"
      >
        Study react hooks (useReducer)
      </span>
    </div>
    <input
      checked={true}
      className="form-check-input p-3 bg-primary rounded-pill  "
      id="flexCheckChecked"
      onChange={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
      type="checkbox"
    />
    <button
      className="btn ms-3 btn-outline-danger  false"
      onClick={[Function]}
    >
      Delete
    </button>
  </li>
</Fragment>
`;
